### PR TITLE
Cleanup PlrHitPlr

### PR DESCRIPTION
**Pull request by AJenbo**

### CODE REVIEW

**Comment by squidcc** [comment]
> As I read it the compiler has now combined the 2*level stuff in to fewer instructions.
' (plr[p]._pLevel<<1) - (plr[pnum]._pLevel<<1)'

File: Source/player.cpp [hunk 1]
```diff
@@ -2781,126 +2781,93 @@ bool __fastcall PlrHitMonst(int pnum, int m)
...
+	hper += plr[pnum]._pIBonusToHit;
+	if ( hper < 5 ) {
+		hper = 5;
+	}
+	if ( hper > 95 ) {
+		hper = 95;
+	}
+
+	int blk;
+	if ( (plr[p]._pmode == 0 || plr[p]._pmode == 4) && plr[p]._pBlockFlag ) {
```
Comment (squidcc) [inline Source/player.cpp]: _pmode == PM_STAND || _pmode == PM_ATTACK

**Comment by squidcc** [comment]
What seems to be happening here is our compiler isn't sure random() won't change the values in plr[pnum] so you have to load them into locals. Maybe this is a sign global optimization was turned on.

**Comment by AJenbo** [comment]
@squidcc looks like your suggestions improved things, I can't look in to it right now, but look like there is one to many calls in the current code and else things looks correct:
image

**Comment by AJenbo** [comment]
Fixed, I just needed to remove one of the calls to random :)

**Comment by seritools** [comment]
@squidcc global optimizations are turned on, it's weird.

**Comment by squidcc** [comment]
Doesn't look like it, I only see /O1 in MakefileVC which is optimize for size. Global optimization would be /Ogs.

**Comment by seritools** [comment]
@squidcc Official MSDN for VC6:

image

**Comment by squidcc** [comment]
Doesn't seem right, frame pointers are definitely still being used in most cases... and /Gf is an insane option to ever turn on, they even got rid of it later on because it was basically a "mess stuff up" switch.

**Comment by seritools** [comment]
Well, I'd be happy to merge changes to the project files and MakefileVC if you got a more accurate switch combination..

**Comment by squidcc** [comment]
I just tried /Ogs on its own and it blew up with
```
fatal error C1001: INTERNAL COMPILER ERROR
(compiler file 'msc1.cpp', line 1794)
```
so I guess that's not really an option.

**Comment by sskras** [comment]
Guys, approximately every second thread I see in the issues is like reading new pages from a captivating, quality detective book.  Thanks!

**Pull request merged**